utils/githooks/find_base.sh

```diff
@@ -23,9 +23,22 @@ if command -v gh > /dev/null 2>&1; then
         fi
     fi
 else
-    # With no 'gh' command installed then check against origin/master.
+    # With no 'gh' command installed, use the "closest" branch as the target,
+    # calculated as the sum of the commits this branch is ahead and behind.
     # shellcheck disable=SC2034
-    TARGET="$ORIGIN"/master
+    all_bases=("master" "release/2.4" "feature/cat_recovery" "feature/multiprovider")
+    TARGET="$ORIGIN/master"
+    min_diff=-1
+    for base in "${all_bases[@]}"; do
+        git rev-parse --verify "$base" 2>/dev/null || continue
+        commits_ahead=$(git log --oneline "$ORIGIN/$base..HEAD" | wc -l)
+        commits_behind=$(git log --oneline "HEAD..$ORIGIN/$base" | wc -l)
+        commits_diff=$((commits_ahead + commits_behind))
+        if [ "$min_diff" -eq -1 ] || [ "$min_diff" -gt "$commits_diff" ]; then
+            TARGET="$ORIGIN/$base"
+            min_diff=$commits_diff
+        fi
+    done
     echo "  Install gh command to auto-detect target branch, assuming $TARGET."
```

Comment:
There's a chance this test could take a while to run, perhaps better to log (most of) this and finish with `checking...` so that if there's a delay in output it's clear what's happening, and how to fix it.

Reply:
FWIW, it's usually pretty fast on my boro node at least
```
$ time git log --oneline origin/release/2.2..HEAD | wc -l
1938

real    0m0.078s
user    0m0.039s
sys     0m0.017s
```
I'm hesitant to add more logging because the hooks are already so verbose with the `ORIGIN` messages